pamelalozano/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -96,6 +93,52 @@
       return eventsMap;
   }
 
+  /*
+  * This function will merge all the occupied times of the mandatory attendees
+  * into a single list of TimeRanges, that way I can create a lit of available 
+  * times that they all have in common.
+  *
+  * 
+  * Runtime: O(p * m) where p is the number of attendees and m the number of events
+  * 
+  * @param mandatoryAttendees   A list of the mandatory attendes
+  * @param eventsMap            A map with the keys as the name of the attendees 
+  *                             and the values as a list with the occupied time ranges
+  *
+  * returns  a list of the available times of the request
+  */
+  public List<TimeRange> checkAvailability(HashSet<String> mandatoryAttendees, HashMap<String, List<TimeRange>> eventsMap){
+    
+    LinkedList<TimeRange> timeOccupied = new LinkedList<>();
+    for (String person : mandatoryAttendees) {
+        List<TimeRange> schedule = eventsMap.get(person);
+        for(TimeRange range : schedule){
+          timeOccupied.add(range);
+       }
+    }
+    //ToDo: Merge times
```

Comment:
I am not sure merging them together is absolutely necessary. I think you can get a list of unoccupied time ranges without doing that step.